src/test/java/edu/ucla/library/iiif/fester/verticles/V3ManifestVerticleTest.java

```diff
@@ -0,0 +1,398 @@
+
+    private static final String POSTCARDS = "capostcards";
+
+    private static final String WORKS = "works";
+
+    @Rule
+    public Timeout myTestTimeout = Timeout.seconds(600);
+
+    private Vertx myVertx;
+
+    private String myRunID;
+
+    private String myJsonFiles;
+
+    /**
+     * Initialization of the testing environment before the tests are run.
+     *
+     * @param aContext A testing context
+     * @throws IOException If there is trouble starting the Vert.x instance
+     */
+    @Before
+    public void setUp(final TestContext aContext) throws IOException {
+        final DeploymentOptions options = new DeploymentOptions();
+        final JsonObject config = new JsonObject();
+        final Async asyncTask = aContext.async();
+
+        myVertx = Vertx.vertx();
+        ConfigRetriever.create(myVertx).getConfig(getConfig -> {
+            if (getConfig.succeeded()) {
+                @SuppressWarnings("rawtypes")
+                final List<Future> futures = new ArrayList<>();
+
+                options.setConfig(config.put(Config.IIIF_BASE_URL, getConfig.result().getString(
+                        Config.IIIF_BASE_URL)).put(Constants.IIIF_API_VERSION, Constants.IIIF_API_V3));
+
+                for (final String verticleName : Arrays.asList(TEST_VERTICLES)) {
+                    final Promise<String> promise = Promise.promise();
+
+                    myVertx.deployVerticle(verticleName, options, deployment -> {
+                        if (deployment.succeeded()) {
+                            promise.complete(deployment.result());
+                        } else {
+                            promise.fail(deployment.cause());
+                        }
+                    });
+
+                    futures.add(promise.future());
+                }
+
+                CompositeFuture.all(futures).onComplete(startup -> {
+                    if (startup.succeeded()) {
+                        final LocalMap<String, String> map = myVertx.sharedData().getLocalMap(Constants.VERTICLE_MAP);
+                        final String deploymentKey = FakeS3BucketVerticle.class.getSimpleName();
+
+                        if (map.containsKey(deploymentKey)) {
+                            try {
+                                myJsonFiles = getS3TempDir(map.get(deploymentKey));
+                                asyncTask.complete();
+                            } catch (final FileNotFoundException details) {
+                                aContext.fail(details);
+                            }
+                        } else {
+                            aContext.fail(LOGGER.getMessage(MessageCodes.MFS_077, deploymentKey));
+                        }
+                    } else {
+
+                    }
+                });
+
+                myRunID = UUID.randomUUID().toString();
+            } else {
+                aContext.fail(getConfig.cause());
+                asyncTask.complete();
```

Comment:
We've not been consistent on this (figuring this is probably a copy and paste), but when you have a guaranteed failure you don't need to complete the task, the failure completes it.